server/jetstream_errors.go

```diff
@@ -0,0 +1,105 @@
+package server
+
+import (
+	"fmt"
+	"strings"
+)
+
+type ErrorIdentifier uint16
+
+// IsNatsErr determines if a error matches ID, if multiple IDs are given if the error matches any of these the function will be true
+func IsNatsErr(err error, ids ...ErrorIdentifier) bool {
+	if err == nil {
+		return false
+	}
+
+	ce, ok := err.(*ApiError)
+	if !ok || ce == nil {
+		return false
+	}
+
+	found := false
+	for _, id := range ids {
+		ae, ok := ApiErrors[id]
+		if !ok || ae == nil {
+			continue
+		}
+
+		if ce.ErrCode == ae.ErrCode {
+			found = true
+		}
+	}
+
+	return found
+}
+
+// ApiError is included in all responses if there was an error.
+type ApiError struct {
+	Code        int    `json:"code"`
+	ErrCode     uint16 `json:"err_code,omitempty"`
+	Description string `json:"description,omitempty"`
+	URL         string `json:"-"`
+	Help        string `json:"-"`
+}
+
+func (e *ApiError) Error() string {
+	return fmt.Sprintf("%s (%d)", e.Description, e.ErrCode)
+}
+
+// ErrOrNewT returns err if it's an ApiError else creates a new error using NewT()
+func (e *ApiError) ErrOrNewT(err error, replacements ...interface{}) *ApiError {
+	if ae, ok := err.(*ApiError); ok {
+		return ae
+	}
+
+	return e.NewT(replacements...)
+}
+
+// ErrOrNew returns err if it's an ApiError else creates a new error
+func (e *ApiError) ErrOrNew(err error) *ApiError {
+	if ae, ok := err.(*ApiError); ok {
+		return ae
+	}
+
+	return e.NewT()
+}
+
+// NewT creates a new error using strings.Replacer on the Description field, arguments must be an even number like NewT("{err}", err)
+func (e *ApiError) NewT(replacements ...interface{}) *ApiError {
```

Comment:
@kozlovic So I added tokenization in 40c9c1c and updated all the things and rebased to master.  wdyt?